pkg/helm/helm.go

```diff
@@ -65,6 +66,58 @@ func NewHelmWrapper(getter genericclioptions.RESTClientGetter, namespace string,
 	}, nil
 }
 
+// Pull
+// inspire by https://github.com/helm/helm/blob/main/cmd/helm/pull.go
+func (h *HelmWrapper) Pull(logger logr.Logger, url string) (chartRequested *chart.Chart, err error) {
+
+	i := action.NewPullWithOpts(action.WithConfig(h.config))
+	h.config.RegistryClient, err = registry.NewClient()
+	if err != nil {
+		logger.Error(err, "cannot create new registry")
+		return nil, err
+	}
+
+	//i.DestDir = "/usr/bin" // destination directory
+	combinedName := strings.Split(url, "/")
+	entryName := combinedName[len(combinedName)-1]
+
+	s := cli.New()
+	s.PluginsDirectory = "testdata/helmhome/helm/plugins"
+	i.Settings = s
+	i.Untar = true
+	i.UntarDir = os.TempDir()
+	i.UntarDir, err = os.MkdirTemp(i.UntarDir, "kubebb-*")
+	if err != nil {
+		logger.Error(err, "Failed to create a new dir")
+		return nil, err
+	}
+
+	_, err = i.Run(url) // chartref - the chart name
+	if err != nil {
+		logger.Error(err, "cannot download chart")
+		return nil, err
+	}
+
+	chartRequested, err = loader.Load(i.UntarDir + "/" + entryName)
+	if err != nil {
+		logger.Error(err, "Cannot load chart")
+		return nil, err
+	}
+
+	if chartRequested.Metadata.Deprecated {
+		logger.V(1).Info("This chart is deprecated")
+	}
+
+	logger.Info(i.UntarDir + "/" + entryName)
+	err = os.RemoveAll(i.UntarDir + "/" + entryName)
```

Comment:
In general, you should delete the folder with `defer` after creating the temporary folder, to prevent an error from occurring in the middle that would cause the logic to not run the delete
some example in helm source code:
```go
if p.Untar {
		var err error
		dest, err = ioutil.TempDir("", "helm-")
		if err != nil {
			return out.String(), errors.Wrap(err, "failed to untar")
		}
		defer os.RemoveAll(dest)
	}
```

Reply:
It seems that they assumed RemoveAll() will not give an error and therefore used defer directly. Can I know if it would be better to create another function cleanFile(String dest) and then defer the function, so that we can log the error?

Reply:
Whether or not to catch errors in `defer` counts as a classic debate in the go community. In go's own source code, it's both.

Reply:
After offline discussion, we decided to directly use defer os.RemoveAll().